src/js/videojs-dash.js

```diff
@@ -116,8 +122,26 @@ class Html5DashJS {
       delete this.player.dash;
     }
   }
+
+  /**
+   * Registers a callback function to be called before the MediaPlayer is initialized.
+   * Does not register duplicates
+   *
+   * @param {Function} fn callback function to register
+   * @function registerInitializationHook
+   */
+  static registerInitializationHook(fn) {
+    const index = Html5DashJS.beforeInitializeHooks_.indexOf(fn);
+
+    if (index <= -1) {
+      // Only add callback if its not a duplicate
+      Html5DashJS.beforeInitializeHooks_.push(fn);
+    }
+  }
 }
 
+Html5DashJS.beforeInitializeHooks_ = [];
```

Comment:
If there are multiple players on the page, won't they all share the same `beforeInitializationHooks_` array?